File: src/ADF.Net.Core/ValueObjects/Paging.cs
﻿using System.Collections.Generic;

namespace ADF.Net.Core.ValueObjects
{
    public class Paging
    {
        public int PageCount { get; set; }
        public int TotalItemCount { get; set; }
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public bool HasPreviousPage { get; set; }
        public bool HasNextPage { get; set; }
        public bool IsFirstPage { get; set; }
        public bool IsLastPage { get; set; }
        public List<KeyValuePair<int, string>> PageSizes { get; set; }
    }
}

File: src/client/web/private/src/app/value-objects/paging.ts
import { IdCodeName } from './id-code-name';

export class Paging {
    pageCount: number;
    totalItemCount: number;
    pageNumber: number;
    pageSize: number;
    hasPreviousPage: boolean;
    hasNextPage: boolean;
    isFirstPage: boolean;
    isLastPage: boolean;
    pageSizes: IdCodeName[];
}
